src/pybind/mgr/cephadm/module.py

```diff
@@ -969,10 +988,16 @@ def _run_cephadm(self,
                         'unable to find python on %s (tried %s in %s)' % (
                             host, remotes.PYTHONS, remotes.PATH))
                 try:
-                    out, err, code = remoto.process.check(
+                    if self.ssh_user == 'root':
+                        out, err, code = remoto.process.check(
                         conn,
                         [python, '-u'],
                         stdin=script.encode('utf-8'))
+                    else:
+                        out, err, code = remoto.process.check(
+                        conn,
+                        ['sudo', python, '-u'],
+                        stdin=script.encode('utf-8'))
```

Comment:
Just add `sudo` to the command (even if the user is root)?

Reply:
No need to do that here. `remoto`'s `Connection` class supports this already. See

https://github.com/alfredodeza/remoto/blob/b08c9ab9a60f660982bdf3464af7e658894ece0a/remoto/backends/__init__.py#L20

plus

https://github.com/ceph/ceph/blob/57734cf105d8e35a47af0e879bb284821d4834e7/src/pybind/mgr/cephadm/module.py#L863-L866